core/src/main/java/brooklyn/entity/rebind/PeriodicDeltaChangeListener.java

```diff
@@ -348,13 +327,40 @@ private void addReferencedObjects(DeltaCollector deltaCollector) {
     }
     
     @VisibleForTesting
-    public void persistNow() {
-        if (!isActive()) {
+    public boolean persistNowSafely(boolean alreadyHasMutex) {
+        Stopwatch timer = Stopwatch.createStarted();
+        try {
+            persistNowInternal(alreadyHasMutex);
+            metrics.noteSuccess(Duration.of(timer));
+            return true;
+        } catch (RuntimeInterruptedException e) {
+            LOG.debug("Interrupted persisting change-delta (rethrowing)", e);
+            metrics.noteFailure(Duration.of(timer));
+            metrics.noteError(e.toString());
+            Thread.currentThread().interrupt();
+            return false;
+        } catch (Exception e) {
```

Comment:
I'd probably merge the `catch Exception` and the `catch Throwable`, with:

        } catch (Throwable t) {
            // Don't rethrow: the behaviour of executionManager is different from a scheduledExecutorService,
            // if we throw an exception, then our task will never get executed again
            LOG.error("Problem persisting change-delta", t);
            metrics.noteFailure(Duration.of(timer));
            metrics.noteError(t.toString());
            Exceptions.propagateIfFatal(t);
            return false;

Downside is that log message is always error, and doesn't include whether or not it is "(rethrowing)".